modules/network/pre-existing-subnetwork/main.tf

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+
+locals {
+  display_name = var.subnetwork_name != null ? var.subnetwork_name : var.subnetwork_self_link
```

Comment:
nit. you can use [`coalesce`](https://developer.hashicorp.com/terraform/language/functions/coalesce) function for better readability.